src/components/molecules/MyContractsTable/index.tsx

```diff
@@ -0,0 +1,160 @@
+import { Grid, Table, TableBody, TableCell, TableHead, TableRow } from "@mui/material";
+import theme from "../../../utils/theme";
+import Typography from "../../atoms/Typography";
+import { IMyContractsTableColumnLabels, IMyContractsTableDataDetail } from "../../../utils/myContracts";
+
+export interface IMyContractsTableProps {
+    tableColumns: IMyContractsTableColumnLabels[];
+    tableData: IMyContractsTableDataDetail[];
+};
+
+const MyContractsTable = ({ tableColumns, tableData }: IMyContractsTableProps) => {
+    return (
+        <Table
+            sx={{
+                width: theme.spacing(249),
+                height: theme.spacing(42),
+                background: theme.palette.common.black
+            }}
+        >
+            <TableHead sx={{ height: theme.spacing(11) }}>
+                <TableRow>
+                    {tableColumns.map((element, index) => (
+                        <TableCell
+                            variant="head"
+                            align="left"
+                            width="20%"
+                            sx={{
+                                pl: theme.spacing(9),
+                            }}
+                            key={index}
+                        >
+                            <Typography
+                                variant="body2"
+                                sx={{ color: theme.palette.text.disabled }}
+                            >
+                                {element.label}
+                            </Typography>
+                        </TableCell>
+                    ))}
+                </TableRow>
+            </TableHead>
+            <TableBody>
+                {tableData.map((element, index) => (
+                    <TableRow key={index} sx={{ height: theme.spacing(15) }}>
+                        <TableCell
```

Comment:
you can add TableCell in a loop as well.  Can you explore about Object.keys which givens the keys of the Object (element in your case).